ride_share.rb

```diff
@@ -0,0 +1,181 @@
+########################################################
+# Step 1: Establish the layers
+
+# In this section of the file, as a series of comments,
+# create a list of the layers you identify.
+# Which layers are nested in each other?
+# Which layers of data "have" within it a different layer?
+# Which layers are "next" to each other?
+    #at the top is all the trips
+    #next layer down is each driver
+    #next layer under each driver is each trip they did
+    #each trip for each driver will be next to each other in the innermost layer
+    #each driver will be next to each other in the middle layer
+    #the "umbrella" layer holds everything from the middle and innermost layer inside of it
+
+
+########################################################
+# Step 2: Assign a data structure to each layer
+
+# Copy your list from above, and in this section
+# determine what data structure each layer should have
+    #outer most layer is an array that will hold all the info
+    #middle layer will be of hashes that belong to each individual driver
+    #inside the hash of each indiv. driver, there is a hash for each trip
+
+########################################################
+# Step 3: Make the data structure!
+
+# Setup the entire data structure:
+# based off of the notes you have above, create the
+# and manually write in data presented in rides.csv
+# You should be copying and pasting the literal data
+# into this data structure, such as "DR0004"
+# and "3rd Feb 2016" and "RD0022"
+
+all_drivers_all_trips = [
+  { DR0001: 
+    [{
+      date_of_trip: "3 Feb 2016",
+      cost: 10,
+      rider_id: "RD0003",
+      rating: 3
+      },
+      { 
+      date_of_trip: "3 Feb 2016",
+      cost: 30,
+      rider_id: "RD0015",
+      rating: 4
+      },
+      {
+      date_of_trip: "5 Feb 2016",
+      cost: 45,
+      rider_id: "RD0003", 
+      rating: 2
+      }]
+  }, 
+
+  {DR0002: 
+    [{
+      date_of_trip: "3 Feb 2016",
+      cost: 25,
+      rider_id: "RD0073",
+      rating:  5
+    }, 
+    {
+      date_of_trip: "4 Feb 2016",
+      cost: 15,
+      rider_id: "RD0013",
+      rating: 1
+    }, 
+    {
+      date_of_trip: "5 Feb 2016",
+      cost: 35,
+      rider_id: "RD0066",
+      rating: 3
+    }]
+  }, 
+
+  {DR0003: 
+    [{
+      date_of_trip: "4 February 2016",
+      cost: 5,
+      rider_id: "RD0066",
+      rating: 5
+    }, 
+    {
+      date_of_trip: "5 Feb 2016",
+      cost: 50,
+      rider_id: "RD0003",
+      rating: 2
+    }]
+  }, 
+
+  {DR0004:
+    [{
```

Comment:
Give these things more newlines. `{` `DR0004` `[` and `{` can all live on their own lines, as it helps with clarity between data structure vs data.